Dockerized

davidhfrankelcodes opened the pull request:
This PR includes a `Dockerfile` and `docker-compose.yaml` file for a docker deployment. Tested successfully on my end.

Thanks! This app is awesome and does exactly what I need! Please consider merging this change, as it provides another way to deploy.

davidhfrankelcodes committed:
Added ability to build image using docker compose build

davidhfrankelcodes committed:
Dockerization instructions in README

ozanmakes commented:
Thanks for this! It looks good and it is nice to have an option for self-hosting. However, I am not 100% sure if `wrangler dev` is the best method for production. Do you use this in production?

Directly using [workerd](https://github.com/cloudflare/workerd) might be the way to go. What do you think?

davidhfrankelcodes commented:
Go for it! I'm not a javascript developer so I don't know. I actually still have this up and running on my homelab in docker with the dockerization in this PR. 

Up to you! I don't know how to implement that.

davidhfrankelcodes committed:
Update and rename docker-compose.yaml to docker-compose-dev.yaml

Changing to dev because I'm not a javascript developer and I don't really know how to do the production deployment.

Enjoyed working on this! Good luck!

-DF

davidhfrankelcodes committed:
Update README.md: Docker deployment is dev only

davidhfrankelcodes committed:
Update docker-compose-dev.yaml: ChatGPT says development is valid and dev is not valid. changed to development

davidhfrankelcodes committed:
Update Dockerfile: Docker deployment is only development

ozanmakes commented:
Merged this. I will investigate further self-hosting options another time. Thanks again!
